sdk/ai/azure-ai-inference/azure/ai/inference/_patch.py

```diff
@@ -482,31 +602,34 @@ def complete(
 
         _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
         _params = kwargs.pop("params", {}) or {}
-        _unknown_params: Union[_models._enums.UnknownParams, None] = None
+        _extra_parameters: Union[_models._enums.UnknownParams, None] = None
 
         content_type: Optional[str] = kwargs.pop("content_type", _headers.pop("Content-Type", None))
 
         if body is _Unset:
             if messages is _Unset:
                 raise TypeError("missing required argument: messages")
             body = {
-                "frequency_penalty": frequency_penalty,
-                "max_tokens": max_tokens,
                 "messages": messages,
-                "model": model,
-                "presence_penalty": presence_penalty,
-                "response_format": response_format,
-                "seed": seed,
-                "stop": stop,
                 "stream": stream,
-                "temperature": temperature,
-                "tool_choice": tool_choice,
-                "tools": tools,
-                "top_p": top_p,
+                "frequency_penalty": frequency_penalty if frequency_penalty is not None else self._frequency_penalty,
+                "max_tokens": max_tokens if max_tokens is not None else self._max_tokens,
+                "model": model if model is not None else self._model,
+                "presence_penalty": presence_penalty if presence_penalty is not None else self._presence_penalty,
+                "response_format": response_format if response_format is not None else self._response_format,
+                "seed": seed if seed is not None else self._seed,
+                "stop": stop if stop is not None else self._stop,
+                "temperature": temperature if temperature is not None else self._temperature,
+                "tool_choice": tool_choice if tool_choice is not None else self._tool_choice,
+                "tools": tools if tools is not None else self._tools,
+                "top_p": top_p if top_p is not None else self._top_p,
```

Comment:
In a scenario where the user set a client default for temperature, but decides they want to clear that default at the method-level by passing in `None`, I'm not sure we'd do the right thing here. Should we be using some sort of sentinel value instead?

Reply:
Per our IM discussion I don't see this is a real scenario to be concerned about. Let's wait and see if there is any user feedback on this.